library/src/test/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionBindingTest.java

```diff
@@ -33,9 +46,275 @@
+        expectedException.expect(NullPointerException.class);
+        expectedException.expectMessage("Section.getItemView() returned null");
+
+        // When
+        sectionAdapter.onCreateViewHolder(null, 12 + SectionedRecyclerViewAdapter.VIEW_TYPE_ITEM_LOADED);
+    }
+
+    @Test
+    public void onCreateViewHolder_withHeaderResourceId_calls_getHeaderViewHolder() {
+        // Given
+        Section headedSection = spy(new SectionImpl(new SectionParameters.Builder(-1).headerResourceId(-2)));
+        sectionAdapter.addSection(headedSection);
+
+        // When
+        sectionAdapter.onCreateViewHolder(null, 12 + SectionedRecyclerViewAdapter.VIEW_TYPE_HEADER);
+
+        // Then
+        verify(section, never()).getHeaderView(any(ViewGroup.class));
+        verify(section, never()).getHeaderViewHolder(any(View.class));
+        verify(headedSection, never()).getHeaderView(any(ViewGroup.class));
+        verify(headedSection, times(1)).getHeaderViewHolder(argThat(hasTag(-2)));
+    }
+
+    @Test
+    public void onCreateViewHolder_withHeaderViewProvided_calls_getHeaderView() {
+        // Given
+        Section providingSection = spy(new SectionImpl(new SectionParameters.Builder(-1).headerViewWillBeProvided()));
+        sectionAdapter.addSection(providingSection); // Third section, view types 12-17
+        doReturn(dummyViewWithTag(42)).when(providingSection).getHeaderView(null);
+
+        // When
+        sectionAdapter.onCreateViewHolder(null, 12 + SectionedRecyclerViewAdapter.VIEW_TYPE_HEADER);
+
+        // Then
+        verify(providingSection, times(1)).getHeaderView(null);
+        verify(providingSection, times(1)).getHeaderViewHolder(argThat(hasTag(42)));
+    }
+
+    @Test
+    public void onCreateViewHolder_withHeaderViewProvided_throws_when_getHeaderView_returns_null() {
+        // Given
+        Section providingSection = spy(new SectionImpl(new SectionParameters.Builder(-1).headerViewWillBeProvided()));
+        sectionAdapter.addSection(providingSection); // Third section, view types 12-17
+        doReturn(null).when(providingSection).getHeaderView(null);
+
+        // Expect exception
+        expectedException.expect(NullPointerException.class);
+        expectedException.expectMessage("Section.getHeaderView() returned null");
+
+        // When
+        sectionAdapter.onCreateViewHolder(null, 12 + SectionedRecyclerViewAdapter.VIEW_TYPE_HEADER);
+    }
+
+    @Test
+    public void onCreateViewHolder_withFooterResourceId_calls_getFooterViewHolder() {
+        // Given
+        Section footedSection = spy(new SectionImpl(new SectionParameters.Builder(-1).footerResourceId(-3)));
+        sectionAdapter.addSection(footedSection);
+
+        // When
+        sectionAdapter.onCreateViewHolder(null, 12 + SectionedRecyclerViewAdapter.VIEW_TYPE_FOOTER);
+
+        // Then
+        verify(section, never()).getHeaderView(any(ViewGroup.class));
+        verify(section, never()).getHeaderViewHolder(any(View.class));
```

Comment:
did you mean to call getFooterView for these

Reply:
Indeed, same above. Fixed now.